### PR TITLE
[NETSTAT] Fix crash when parsing the protocol CORE-19005

### PR DESCRIPTION
**Pull request by JoachimHenze**
## Purpose

Fix a crash that happens when accessing cmd args out of bounds and show usage help.

JIRA issue: [CORE-19005](https://jira.reactos.org/browse/CORE-19005)

The crash was a regression of 0.4.11-dev-814-g 2b5507336017cdea726f146a225823a932df54c2

Before my fix:
https://jira.reactos.org/secure/attachment/65929/65929_0.4.15-dev-6172-gc63489f_netstatCrash_abnop.png

After my quick fix: (I showed usage help then, no crash anymore. Wrong screen output, wrong error level)
https://jira.reactos.org/secure/attachment/65931/65931_quick_fix.webm

After my final fix: (no crash, proper screen output, proper error-level)
https://jira.reactos.org/secure/attachment/65933/65933_put_love_into_the_fix_to_exactly_match_windows_screenoutput_and_errorlevel.webm

### CODE REVIEW

File: base/applications/network/netstat/netstat.c [hunk 1]
```diff
@@ -103,6 +100,8 @@ BOOL ParseCmdline(int argc, wchar_t* argv[])
                         break;
                     case L'p':
                         bDoShowProtoCons = TRUE;
+                        if(i+1 >= argc)
```
Comment (katahiromz) [inline base/applications/network/netstat/netstat.c:103]:
```suggestion
                        if (i + 1 >= argc)
```

Comment (JoachimHenze) [reply base/applications/network/netstat/netstat.c]: Lol, I already anticipated that and just fixed it.

File: base/applications/network/netstat/netstat.c [hunk 2]
```diff
@@ -248,7 +243,7 @@ VOID ShowIpStatistics(VOID)
     PMIB_IPSTATS pIpStats;
     DWORD dwRetVal;
 
-    pIpStats = (MIB_IPSTATS*) HeapAlloc(GetProcessHeap(), 0, sizeof(MIB_IPSTATS));
+    pIpStats = (MIB_IPSTATS*)HeapAlloc(GetProcessHeap(), 0, sizeof(MIB_IPSTATS));
```
Comment (HBelusca) [inline base/applications/network/netstat/netstat.c:246]: Missing NULL checks by the way...

File: base/applications/network/netstat/netstat.c [hunk 3]
```diff
@@ -609,10 +602,8 @@ GetIpHostName(BOOL Local, UINT IpAddr, CHAR Name[], INT NameLen)
 }
 
 /*
- *
  * Parse command line parameters and set any options
- * Run display output, looping over set intervals if a number is given
- *
+ * Run display output, looping over set interval if a number is given
```
Comment (HBelusca) [inline base/applications/network/netstat/netstat.c:606]:
```suggestion
 * Run display output, looping over set interval if a number is given.
```

File: base/applications/network/netstat/netstat.c [hunk 4]
```diff
@@ -609,10 +602,8 @@ GetIpHostName(BOOL Local, UINT IpAddr, CHAR Name[], INT NameLen)
 }
 
 /*
- *
  * Parse command line parameters and set any options
```
Comment (HBelusca) [inline base/applications/network/netstat/netstat.c:605]:
```suggestion
 * Parse command line parameters and set any options.
```

File: base/applications/network/netstat/netstat.c [hunk 5]
```diff
@@ -374,12 +368,12 @@ VOID ShowEthernetStatistics(VOID)
     DWORD dwSize = 0;
     DWORD dwRetVal = 0;
 
-    pIfTable = (MIB_IFTABLE*) HeapAlloc(GetProcessHeap(), 0, sizeof(MIB_IFTABLE));
+    pIfTable = (MIB_IFTABLE*)HeapAlloc(GetProcessHeap(), 0, sizeof(MIB_IFTABLE));
```
Comment (KRosUser2) [inline base/applications/network/netstat/netstat.c:371]: NULL check is missing.

File: base/applications/network/netstat/netstat.c [hunk 6]
```diff
@@ -374,12 +368,12 @@ VOID ShowEthernetStatistics(VOID)
...
 
     if (GetIfTable(pIfTable, &dwSize, 0) == ERROR_INSUFFICIENT_BUFFER)
     {
         HeapFree(GetProcessHeap(), 0, pIfTable);
-        pIfTable = (MIB_IFTABLE*) HeapAlloc(GetProcessHeap(), 0, dwSize);
+        pIfTable = (MIB_IFTABLE*)HeapAlloc(GetProcessHeap(), 0, dwSize);
```
Comment (KRosUser2) [inline base/applications/network/netstat/netstat.c:376]: NULL check is missing.

File: base/applications/network/netstat/netstat.c [hunk 7]
```diff
@@ -418,17 +412,17 @@ BOOL ShowTcpTable(VOID)
...
 
     /* Get the table of TCP endpoints */
-    dwSize = sizeof (MIB_TCPTABLE_OWNER_PID);
+    dwSize = sizeof(MIB_TCPTABLE_OWNER_PID);
     /* Should also work when we get new connections between 2 GetTcpTable()
      * calls: */
     do
     {
-        tcpTable = (PMIB_TCPTABLE_OWNER_PID) HeapAlloc(GetProcessHeap(), 0, dwSize);
+        tcpTable = (PMIB_TCPTABLE_OWNER_PID)HeapAlloc(GetProcessHeap(), 0, dwSize);
```
Comment (KRosUser2) [inline base/applications/network/netstat/netstat.c:420]: NULL check is missing.

File: base/applications/network/netstat/netstat.c [hunk 8]
```diff
@@ -498,7 +492,7 @@ BOOL ShowUdpTable(VOID)
         DoFormatMessage(error);
         return FALSE;
     }
-    udpTable = (PMIB_UDPTABLE_OWNER_PID) HeapAlloc(GetProcessHeap(), 0, dwSize);
+    udpTable = (PMIB_UDPTABLE_OWNER_PID)HeapAlloc(GetProcessHeap(), 0, dwSize);
```
Comment (KRosUser2) [inline base/applications/network/netstat/netstat.c:495]: NULL check is missing.

**Review by KRosUser2 — CHANGES_REQUESTED** [review]
Several NULL checks are missing

File: base/applications/network/netstat/netstat.c (hunk 6 above)
Comment (JoachimHenze) [reply base/applications/network/netstat/netstat.c:376]: This is not scope of this ticket.

File: base/applications/network/netstat/netstat.c (hunk 5 above)
Comment (JoachimHenze) [reply base/applications/network/netstat/netstat.c:371]: This is not scope of this ticket.

File: base/applications/network/netstat/netstat.c (hunk 4 above)
Comment (JoachimHenze) [reply base/applications/network/netstat/netstat.c:605]: No, I didn't touch that line, and I don't want to add that.

File: base/applications/network/netstat/netstat.c (hunk 3 above)
Comment (JoachimHenze) [reply base/applications/network/netstat/netstat.c:606]: No, I don't want to add that.

File: base/applications/network/netstat/netstat.c (hunk 2 above)
Comment (JoachimHenze) [reply base/applications/network/netstat/netstat.c:246]: This is not scope of this ticket.

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]
Maybe use shorter goto label name, like `goto skip;` instead of `goto StopParsingAndShowUsageHelp;`

**Comment by learn-more** [comment]
> Maybe use shorter goto label name, like `goto skip;` instead of `goto StopParsingAndShowUsageHelp;`

Why?

**Comment by binarymaster** [comment]
A label with a name this long looks strange to me.

**Comment by learn-more** [comment]
> A label with a name this long looks strange to me.

I prefer longer readable labels over shorter abbreviations

**Pull request merged**